target/linux/ipq40xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata

```diff
@@ -40,6 +40,10 @@ case "$FIRMWARE" in
 		# OEM assigns 4 sequential MACs
 		ath10k_patch_mac $(macaddr_setbit_la $(macaddr_add "$(cat /sys/class/net/eth0/address)" 4))
 		;;
+	linksys,whw03)
+		caldata_extract_mmc "0:ART" 0x9000 0x2f20
```

Comment:
> Wait, NVMEM now supports MMC devices as well so all of this caldata loading can be moved to NVMEM. For example look at: [49ed52b](https://github.com/openwrt/openwrt/commit/49ed52b862c9f3e0dc1c1024ccd4db3bc9795122) and [bf20585](https://github.com/openwrt/openwrt/commit/bf205854128cc791e5805ed20ef3bd0bb53c339a)

@robimarko 

NVMEM:

not that i would want to do this now, because before converting caldata and mac extraction to nvmem, first i would need to have macs sorted out, and there are outstanding mac issues in the OP.

besides that, AFAICT the links you provided do not really help:
- the mac address base is stored in env-layout, not fixed-layout. your samples extract macs from fixed-layout.
- to obtain actual macs, the mac address base may need to be altered in ways still to be determined.
- the caldata needs to be patched with macs. your samples do extract caldata from mmc, but do not patch them.

another problem is that i can't keep this borrowed V1 forever. i need to reassemble it and return it sooner or later, and i haven't gotten any help with my questions in the OP. if you know someone that can help me, please try getting their attention.

thank you very much!

Reply:
Yeah, it all boils down to the fact that we lack support for reading the MAC from U-Boot like key/value pair storage with voluntary naming (It only works with ethNaddr) style.

So, ignore my NVMEM comments for now.